### PR TITLE
feat: implement upgrade component command

### PR DESCRIPTION
**Pull request by ramabit**
#5

### CODE REVIEW

**Comment by anoack93** [comment]
consider adding the command information to our README

File: src/utilities.js [hunk 1]
```diff
@@ -48,10 +58,21 @@ function validateKebabCaseName (name) {
   return name
 }
 
+/**
+ * @param {string} destinationPath
+ */
+function validateDirectoryExists(destinationPath) {
+  if (!existsSync(destinationPath)) {
+    throw new Error(`'${destinationPath}' does not exist, please create the component directly`)
```
Comment (anoack93) [inline src/utilities.js:66]: improve message here

File: src/gulpfile.js [hunk 2]
```diff
@@ -33,19 +61,50 @@ function generateComponentFiles (fullTemplatePath, componentPath, name, componen
...
+    throw new Error(`flavour '${effectiveFlavour}' does not exist, choose one of: ${availableFlavours}`)
+  }
+
+  const existingFiles = getFiles(destinationPath)
+  const existingFileExtensions = existingFiles.map(fileName => fileName.split(upperCamelCaseName + '.')[1])
+
+  const templateFiles = getFiles(path.join(fullTemplatePath, effectiveFlavour, 'ComponentTemplate'))
+  const templateFilesExtensions = templateFiles.map(fileName => fileName.split('ComponentTemplate.')[1])
+
+  templateFilesExtensions.forEach((extension) => {
```
Comment (anoack93) [inline src/gulpfile.js:95]: we should also support files without ComponentTemplate naming convention like README.md

**Review by anoack93 — CHANGES_REQUESTED** [review]

File: src/gulpfile.js (hunk 2 above)
Comment (ramabit) [reply src/gulpfile.js]: done

File: src/utilities.js (hunk 1 above)
Comment (ramabit) [reply src/utilities.js]: done

**Comment by ramabit** [comment]
@anoack93 please check again

**Review by anoack93 — APPROVED** [review]

**Pull request merged**